### ISSUE TITLE
Add "Linear" MGLInterpolationMode

### ISSUE THREAD

**Issue by ericrwolfe** [mapbox/mapbox-gl-native#8457]
[`MGLInterpolationMode`](https://www.mapbox.com/ios-sdk/api/3.5.0-rc.1/Enums/MGLInterpolationMode.html) is rather confusing, particularly since only one of the options actually deals with interpolation:

- `exponential` (values interpolated)
- `interval` (_not_ interpolated)
- `categorical` (_not_ interpolated)
- `identity` (_not_ interpolated)

90% of developers are going to be using `+[MGLStyleValue valueWithInterpolationMode:cameraStops:options:]` and will be looking for linear interpolation as the default. Right now, I naturally reach for `interval` given the options, so I propose adding another `linear` option for clarity: 

- __`linear`__ (values interpolated; alias for exponential)
- `exponential` (values interpolated)
- `interval` (_not_ interpolated)
- `categorical` (_not_ interpolated)
- `identity` (_not_ interpolated)

This could be a simple alias for `exponential`, with the default `MGLStyleFunctionOptionInterpolationBase` value of 1.

/cc @boundsj @jmkiley @friedbunny

**Comment by boundsj** [comment]
@ericrwolfe I wanted to check in again here to see how much of a pain point this still is now that the API has existed for awhile and documentation and examples have improved.

I think what is proposed here would lead to an API that allows for something like:

```Swift
circleStyleLayer.circleColor = MGLStyleValue<MGLColor>(
  interpolationMode: .linear,
  cameraStops: stops
)
```

as an alias for:

```Swift
// using an interpolation mode of exponential and passing no options results in linear interpolation
circleStyleLayer.circleColor = MGLStyleValue<MGLColor>(
  interpolationMode: .exponential,
  cameraStops: stops
)
```

Implementing this API would be trivial. Updating the tests, manual docs, and generated docs would not take that much longer. However, this API would be in addition to the [style spec](https://www.mapbox.com/mapbox-gl-js/style-spec/#function-type) and exist only on one platform. So, I think it is worth checking in again to see if this is really required just on iOS and macOS.

Would it be possible to solve this problem by adding something to the relevant [existing docs](https://www.mapbox.com/ios-sdk/api/3.5.4/Enums/MGLInterpolationMode.html#/c:@E@MGLInterpolationMode@MGLInterpolationModeExponential) or add some new documentation?

cc @friedbunny @jmkiley

**Comment by ericrwolfe** [comment]
> However, this API would be in addition to the style spec and exist only on one platform. So, I think it is worth checking in again to see if this is really required just on iOS and macOS.

@boundsj I'd be okay with revisiting this in a future API redesign if cross-platform inconsistency is a concern. 

I'm not sure I'm aware of any written issues with developers getting tripped up on this (cc @jmkiley), though I'd be interested in user testing some of the more complex APIs with developers at some point.

**Comment by 1ec5** [comment]
Taking off the v3.6.0 milestone since this isn’t going to get fixed in time. Additionally, the entire style function API will be redesigned as part of the expressions feature: #8074.

**Comment by 1ec5** [comment]
As part of the expressions implementation in mapbox/mapbox-gl-js#4777, the `curve` operator accepts a `linear` interpolation type equivalent to `exponential` with a base of 1. Therefore, the suggestion above will be implemented in the course of implementing #8074.

**Comment by 1ec5** [comment]
Implemented in #8074 for iOS SDK v4.0.0 and macOS SDK v0.7.0.
